lib/full-site-editing/edit-site-page.php

```diff
@@ -77,51 +77,24 @@ function gutenberg_get_editor_styles() {
 }
 
 /**
- * Initialize the Gutenberg Templates List Page.
- *
- * @param array $settings The editor settings.
+ * Get the Gutenberg Templates List Page preload paths.
  */
-function gutenberg_edit_site_list_init( $settings ) {
-	wp_enqueue_script( 'wp-edit-site' );
-	wp_enqueue_style( 'wp-edit-site' );
-	wp_enqueue_media();
+function gutenberg_edit_site_list_preload_paths() {
+	if ( ! gutenberg_is_edit_site_list_page() ) {
+		return array();
+	}
 
 	$post_type = get_post_type_object( $_GET['postType'] );
 
 	if ( ! $post_type ) {
 		wp_die( __( 'Invalid post type.', 'gutenberg' ) );
 	}
 
-	$preload_data = array_reduce(
-		array(
-			'/',
-			"/wp/v2/types/$post_type->name?context=edit",
-			'/wp/v2/types?context=edit',
-			"/wp/v2/$post_type->rest_base?context=edit&per_page=-1",
-		),
-		'rest_preload_api_request',
-		array()
-	);
-
-	wp_add_inline_script(
-		'wp-api-fetch',
-		sprintf(
-			'wp.apiFetch.use( wp.apiFetch.createPreloadingMiddleware( %s ) );',
-			wp_json_encode( $preload_data )
-		),
-		'after'
-	);
-
-	wp_add_inline_script(
-		'wp-edit-site',
-		sprintf(
-			'wp.domReady( function() {
-				wp.editSite.initializeList( "%s", "%s", %s );
-			} );',
-			'edit-site-editor',
-			$post_type->name,
-			wp_json_encode( $settings )
-		)
+	return array(
+		'/',
+		"/wp/v2/types/$post_type->name?context=edit",
+		'/wp/v2/types?context=edit',
+		"/wp/v2/$post_type->rest_base?context=edit&per_page=-1",
```

Comment:
We probably need to preload everything instead, so that navigating to the other list page is smooth on first load. Or we can implement a loading state.